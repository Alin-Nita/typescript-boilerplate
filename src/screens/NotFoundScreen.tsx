import { Link } from "react-router-dom";
import Layout from "../components/Layout";
import { ErrorCode } from "../enums/ErrorCode";

const NotFoundScreen = () => (
  <Layout>
    <h1>Error {ErrorCode.NotFound} - Not Found!</h1>
    <Link to="/">Go Home</Link>
  </Layout>
);

export default NotFoundScreen;
