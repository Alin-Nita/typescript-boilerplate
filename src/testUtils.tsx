import { render, RenderResult } from "@testing-library/react";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import App from "./App";

export const customRender = (
  ui: JSX.Element,
  options = { useRouting: true }
) => {
  let uiResult = ui;

  if (options.useRouting) {
    uiResult = wrapWithRouting(ui);
  }

  return render(uiResult);
};

const wrapWithRouting = (ui: JSX.Element): JSX.Element => {
  return <BrowserRouter>{ui}</BrowserRouter>;
};

// const AllTheProviders = ({ children }: { children: any }) => {
//   return <BrowserRouter>{children}</BrowserRouter>;
// };

// const customRender = (ui: any, options: any) =>
//   render(ui, { wrapper: AllTheProviders, ...options });
