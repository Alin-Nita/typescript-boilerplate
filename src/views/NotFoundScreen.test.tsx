import { render } from "@testing-library/react";
import NotFoundScreen from "./NotFoundScreen";
import { BrowserRouter } from "react-router-dom";
import { customRender } from "../testUtils";

export default describe("NotFoundScreen", () => {
  it("Renders", () => {
    const { container } = customRender(<NotFoundScreen />);

    expect(container).toMatchSnapshot();
  });
});
