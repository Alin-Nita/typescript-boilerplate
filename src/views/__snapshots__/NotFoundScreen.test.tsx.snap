// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundScreen Renders 1`] = `
<div>
  <div
    class="layout"
  >
    <main
      class="layout__main"
    >
      <h1>
        Error 
        404
         - Not Found!
      </h1>
      <a
        href="/"
      >
        Go Home
      </a>
    </main>
  </div>
</div>
`;
